src/mdlComponentHandler.js

```diff
@@ -342,7 +342,7 @@ componentHandler = (function() {
    * @param {?componentHandler.Component} component
    */
   function deconstructComponentInternal(component) {
-    var componentIndex = createdComponents_.indexOf(component);
+    var componentIndex = createdComponents_.indexOf(/** @type {componentHandler.Component} */ (component));
```

Comment:
Not always. Just in cases like this, where you need to "cast" to a more specific type than the one passed into the method.